### ISSUE TITLE
The Waku Whitepaper

### ISSUE THREAD

**Issue by alrevuelta** [waku-org/research#7]
After some internal discussions, we thought it would be a good idea to write **The Waku Whitepaper**, explaining:
* **What** Waku is.
* **Which** problem it's solving.
* **How** it's solving it.

The idea would be to expand [this paper](https://arxiv.org/ftp/arxiv/papers/2207/2207.00038.pdf), being more extensive and specific, reflecting what we aim waku to be. Note that this is not a spec or RFC. It should explain how the proposed problem would be solved but without deep technical implementation details. It should have an extension of around 15 pages, written with academic rigor and somewhat opinionated.

This paper would serve as a public commitment on what we want waku to be (like [Ulysses](https://en.wikipedia.org/wiki/The_Sirens_and_Ulysses) tying himself to the ship). It will also help to maintain academic presence and would also serve to explain new hires, donors, grants or investors what we are building.

I would suggest the following structure:
* Introduction: communication protocols, architectures, client-server, trust, decentralization, problems.
* History: from whisper to waku v2.
* State of the art: what has been done in the field, and where we stand.
* State the problem we want to solve: modular decentralized censorship resistant p2p communications protocol.
* Explain how waku is solving the problem:
  * Peer management and discovery: DHTs, peer discovery, capabilities, needle in a haystack, peer management techniques.
  * Relay protocol: overlay networks, theoretical latencies, scalability with sharding, bandwidth limitations, spam protecion, RLN.
  * Store protocol: req/resp domain, synchronization, trust assumptions.
  * Light protocols: explain light-publish/light-subscribe, trust assumptions and use cases.
  * Incentives: from crypto economic to other game theoretical ones.
  * Dimensioning: Provide theoretical analysis to proof that such protocol is feasible, and set some "physical limits" on what can be archived.
  * Tradeoffs: Clearly explain the trade-offs that exist in such system and where we stand. See trilemma (mgssize, msgrate, bw)
  * Blockchain dependencies: With RLN waku will piggy back on a blockchain. Implications of this, light protocols, trust, etc.

(+ ... other things I'm missing)

Some whitepapers we can use for reference:
* https://gavwood.com/paper.pdf
* https://arxiv.org/pdf/1905.09274.pdf
* https://assets.polkadot.network/Polkadot-whitepaper.pdf

- [x] Approval required before moving on.

**Comment by alrevuelta** [comment]
@fryorcraken @jm-clius can I get some preliminary feedback on the idea?

**Comment by arnetheduck** [comment]
:+1:

**Comment by jm-clius** [comment]
Yes, agree on direction/motivation here and think the scope looks good.

**Comment by fryorcraken** [comment]
Yes looks good.
Would add notion of adaptive node to the list of what needs to be covered.

**Comment by chaitanyaprem** [comment]
Scope is good, One point i feel we should definitely try to indicate/highlight is how the minimum node required to support the network is lightweight(which furthers decentralization). I guess it would get covered as part of adaptive node, but would be good to indicate this thought process as well.

A nice section to include would be `what Waku is not trying to solve/out of scope of its domain`. It would set a reader's scope and wouldn't leave for open assumptions.
